api/application.go

```diff
@@ -388,18 +390,20 @@ func (h ApplicationHandler) Update(ctx *gin.Context) {
 		}
 	}
 
-	rtx := RichContext(ctx)
-	tr := trigger.Application{
-		Trigger: trigger.Trigger{
-			TaskManager: rtx.TaskManager,
-			Client:      rtx.Client,
-			DB:          h.DB(ctx),
-		},
-	}
-	err = tr.Updated(m)
-	if err != nil {
-		_ = ctx.Error(err)
-		return
+	if r.Repository != nil {
```

Comment:
How about moving the check for the repository into the trigger.Application?  When the repository is nil, the trigger could no-op.